tests/test_dataset_factory.py

```diff
@@ -24,6 +25,20 @@ def dataset_factory(self):
                                                                         batch_size=self.batch_size)
         yield train_loader, valid_loader, test_loader
 
+    def test_dataset_last_batch_has_proper_number_of_examples(self, dataset_factory):
+        _, valid_loader, _ = dataset_factory
+        valid_data, valid_labels = next(iter(valid_loader))
+        _, first_batch_counts = np.unique(valid_labels, return_counts=True)
+        while True:
+            try:
+                valid_data, valid_labels = next(dataloader_iterator)
```

Comment:
Le linting a pas passé à cause de cette ligne la! (dataloader_iterator n'existe pas)

Très cool de faire un test pour ça btw! Mais je ne vois pas comme ça va passer par contre. C'est normale qu'il y aille moins d'example dans la dernière batch si le nombre total d'examples n'est pas un multiple de la batch size. Aussi, la tu check que la premiere & la derniere batch a exactement le même nombre de labels. J'pense pas que ce soit requis qu'il y aille exactement le même nombre de labels dans chaque batch. 

Dans le fond pour valider #17, on peut juste créer deux datasets et valider qu'ils ont le même nombre d'examples. Le seed est set dans main.py, alors il va falloir setter le seed dans le test, sinon je ne pense pas que ça va passer.

Reply:
![image](https://user-images.githubusercontent.com/6471460/114088013-f21fa100-988a-11eb-9cdc-e64a57d3aafa.png)
See the comments in code. Basically, the len of the datasets are always the same and the last batch of valid is always of a different shape no matter if the seed is set or not.